cmake/SDL.cmake

```diff
@@ -68,7 +68,7 @@ if(UNIX)
         sdl_gnu_common_ccxx_flags(ONEDNN_SDL_COMPILER_FLAGS CMAKE_CXX_COMPILER_VERSION)
         sdl_gnu_src_ccxx_flags(CMAKE_SRC_CCXX_FLAGS)
         sdl_gnu_example_ccxx_flags(CMAKE_EXAMPLE_CCXX_FLAGS)
-    elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
I assume it's done to cover the AppleClang case. I think we should be more specific on what's allowed to show the intent. To do so we could use the following regex to cover Clang and AppleClang: `^(Clang|AppleClang)$` throughout the build system

By the way, it seems that there is one place in the build system where we already handle such a case:
https://github.com/oneapi-src/oneDNN/blob/c852fdc60073a1ae298618c4faf26487b4144755/cmake/coverage.cmake#L39
I think either regex will do.